bugwarrior/services/bz.py

```diff
@@ -260,9 +276,36 @@ def issues(self):
 
                 extra['needinfo_since'] = pytz.UTC.localize(mod_date)
 
+            if issue['status'] == 'ASSIGNED':
+                extra['assigned_on'] = self._get_assigned_date(issue)
+            else:
+                extra['assigned_on'] = None
+
             issue_obj.update_extra(extra)
             yield issue_obj
 
+    def _get_assigned_date(self, issue):
+        assigned_date = None
+
+        if issue['status'] != 'ASSIGNED':
+            return assigned_date
+
+        bug = self.bz.getbug(issue['id'])
+        history = bug.get_history()['bugs'][0]['history']
+
+        # this is already in chronological order, so the last change is the one we want
+        for h in history:
+            for change in h['changes']:
+                if change['field_name'] == 'status' and change['added'] == 'ASSIGNED':
+                  assigned_date = h['when']
```

Comment:
Why can't we just say `h = history[-1]` rather than looping?

Reply:
It looks like this comment was taken into account, but now the code only looks at the last update to the bug, and if *that* changed it to 'ASSIGNED', then records assigned_date.  If the last update is something else, for example adding a comment, then I think it won't find it.